### PR TITLE
feat: 🎉 added source `https://www2.kickassanime.ro`

### LINKED ISSUE #502: [Source Request] https://www2.kickassanime.ro/
### Summary

Source Request for https://www2.kickassanime.ro/

### The Problem

N/A

### The Ideal Solution

N/A

### The Current Solution

_No response_

### Additional Context

Please add as additional source for https://www2.kickassanime.ro/

### PR DESCRIPTION
**Pull request by TobyBridle**
Added source "https://www2.kickassanime.ro"

Servers:
- [ ] Sapphire Duck (Not planned due to addition of Cronchy)
- [x] Pink Bird
- [x] StreamSB
- [x] Maverickki (**Soft Subbed**)
- [ ] VidStreaming
- - [x] VidStreaming[StreamSB]
- - [ ] Others (e.g Doodstream)
- [x] BetaPlayer
- [x] Gogo
- [x] VidCDN
- [x] AddKAA
- [x] KickassAnimeV2
- [x] Streamtape

closes #502

### CODE REVIEW

File: app/src/main/java/ani/saikou/parsers/AnimeSources.kt [hunk 1]
```diff
@@ -12,6 +12,7 @@ object AnimeSources : WatchSources() {
         "Marin" to ::Marin,
         "AnimePahe" to ::AnimePahe,
         "ConsumeBili" to ::ConsumeBili,
+        "KickAss" to ::KickassAnime
```
Comment (scrazzz) [inline app/src/main/java/ani/saikou/parsers/AnimeSources.kt:15]:
```suggestion
        "KickAssAnime" to ::KickAssAnime
```

File: app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+import android.net.Uri
+import android.util.Base64
+import ani.saikou.*
+import ani.saikou.parsers.*
+import ani.saikou.parsers.anime.extractors.*
+import java.net.URLDecoder
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+class KickassAnime : AnimeParser() {
```
Comment (scrazzz) [inline app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt:12]:
```suggestion
class KickAssAnime : AnimeParser() {
```

File: app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+
+    override val name: String = "Kickass Anime"
+    override val saveName: String = "kickass_anime"
```
Comment (scrazzz) [inline app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt:15]:
```suggestion
    override val saveName: String = "kick_ass_anime"
```

File: app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+            // www2.kickassanime.ro/uploads)
+        )
+    }
+
+    @Serializable
+    data class KAASTLink(val episode: KAASTObject, val ext_servers: MutableList<ExternalServer>) {
+
+        @Serializable
+        data class KAASTObject(
+            //        val name: String,
```
Comment (scrazzz) [inline app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt:286]: These comments could be cleaned up if you're not using it.

File: app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+import ani.saikou.*
+import ani.saikou.parsers.*
+import ani.saikou.parsers.anime.extractors.*
+import java.net.URLDecoder
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+class KickassAnime : AnimeParser() {
+
+    override val name: String = "Kickass Anime"
```
Comment (scrazzz) [inline app/src/main/java/ani/saikou/parsers/anime/KickassAnime.kt:14]:
```suggestion
    override val name: String = "KickAssAnime"
```

**Review by scrazzz — CHANGES_REQUESTED** [review]
LGTM. There's a few minor things that I've suggested before this gets merged.

**Comment by TobyBridle** [comment]
All done. There's nothing I plan on changing/adding so I'm happy for this to be merged now

**Comment by brahmkshatriya** [comment]
Can you seperate out the video extractors to seperate files?

**Comment by TobyBridle** [comment]
> Can you seperate out the video extractors to seperate files?

Sure, I had that originally but was told that because none of the other sources used them to keep them in the same file.
I'll change it now and push again

**Review by scrazzz — APPROVED** [review]

File: app/src/main/java/ani/saikou/parsers/AnimeSources.kt (hunk 1 above)
Comment (scrazzz) [reply app/src/main/java/ani/saikou/parsers/AnimeSources.kt]: This hasn't been resolved yet in the code.

**Comment by scrazzz** [comment]
LGTM. Thanks.

**Pull request merged**